### DISCUSSION TITLE
Feature request: command to show time running and time missing

### DISCUSSION THREAD

**Discussion by wAuner** [jotaen/klog#390] [Feature ideas & discussions]
I propose a command like `klog status` (feel free to come up with a better name) that does these things:
When the user makes an entry with an open ended timeframe like `08:00 - ?` then `klog status` returns the current running time from `08:00 - now`. 

If additionally a should time is provided, it also reports back the delta to the should time. e.g. 8h! and it is 14:00, -45min, then it would say "-2h45m" and ideally also the time when the should time would be reached, 16:45 in this case.

**Comment by jotaen** [comment]
Do you know `klog total --today --live --diff`?

It’s admittedly a bit verbose, so a “shortcut” would be nice indeed. Plus, `--live` is blocking at the moment, and I guess that should be optional.

So maybe something along the lines of `klog status` and `klog status --live`.

**Comment by wAuner** [comment]
thanks for the hint, I haven't used that before. Yes, something like this, but added with the ETA when the should time will be reached.

Also note: There seems to be a bug:
`klog total --today --diff`
Does not "respect" the current time and seems to ignore the open range while `klog total --today --diff --live` does, but maybe that is intentional. 

But what should not happen is that `--live` spams the terminal buffer like this:
```
Total: 5h13m
Should: 8h!
Diff: -2h47m
(In 1 record)
Total: 5h13m
Should: 8h!
Diff: -2h47m
(In 1 record)
Total: 5h13m
Should: 8h!
Diff: -2h47m
(In 1 record)
Total: 5h13m
Should: 8h!
Diff: -2h47m
(In 1 record)
Total: 5h13m
Should: 8h!
Diff: -2h47m
(In 1 record)
Total: 5h13m
Should: 8h!
Diff: -2h47m
(In 1 record)
Total: 5h15m
Should: 8h!
Diff: -2h45m
(In 1 record)
q
Total: 5h15m
Should: 8h!
Diff: -2h45m
(In 1 record)
```

**Comment by jotaen** [comment]
It’s intentional that `total` doesn’t care about open ranges. It’s supposed to be “deterministic”, so it shouldn’t be dependent on the time of execution. (That’s also how open ranges are documented, btw.) I guess this is another sign that the regular `total` and this status check you proposed should be two different commands. Otherwise it’s confusing when some flag fundamentally changes the behaviour.

Also the ETA time is a really useful idea.

Good catch about the buffer spamming. Is this a verbatim copy? On my terminal the buffer is also constantly filled, but the cursor gets reset to position `0,0` so that it appears as a static, ongoing screen. Will look into this, I probably should be using a different ANSI sequence.

(**edit** I felt so free to edit your comment above to shorten the output a bit, so that this page is easier to scroll 😄 I hope you don’t mind that)

**Comment by wAuner** [comment]
yeah, shortening made sense :)
While the command is running it appears to stand still. But when I exit and scroll back I see the constant "spamming".
Maybe you could use `\r`

**Comment by jotaen** [comment]
`klog now` will land in v1.3 🎉 

I thought about names back and forth, btw.: `status` was too generic, `check` potentially misleading, `checkin` too long and meaningless, etc.
I like `now`, because it’s short and it communicates exactly was this command is about: to check how the times would be **right now**. With `--follow` you can keep it open in the shell – this time without buffer spamming 👀 

<img width="470" alt="Screenshot 2021-02-13 at 21 06 33" src="https://user-images.githubusercontent.com/3618384/107860462-bb577c80-6e3f-11eb-8c41-40bd6c41b370.png">
